test/integration/template_database_test.go

```diff
@@ -251,6 +251,60 @@ func TestDatabaseNodePortServiceRenders(t *testing.T) {
 	}
 }
 
+func TestDatabaseInvalidArchiveType(t *testing.T) {
+	// Path to the helm chart we will test
+	helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"cloud.provider":                        "amazon",
+			"database.te.externalAccess.enabled":    "true",
+			"database.te.externalAccess.type":       "NodePort",
+			"database.te.externalAccess.internalIP": "true",
+			"database.archiveType": "invalid",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "release-name", []string{"templates/service.yaml"})
+	assert.Contains(t, err.Error(), "database.archiveType: Does not match pattern")
```

Comment:
Can you post the full error? Does Helm print the pattern etc.?

Reply:
with the code change to an enum, the new error check is `database.archiveType must be one of the following:` - I didn't include the allowed values in the check to prevent a test failure if new values are added in the future.